specutils/io/parsing_utils.py

```diff
@@ -99,18 +99,18 @@ def spectrum_from_column_mapping(table, column_mapping, wcs=None):
             kwarg_val = u.Quantity(table[col_name], tab_unit)
 
             # Attempt to convert the table unit to the user-defined unit.
-            log.debug("Attempting auto-convert of table unit '%s' to "
-                          "user-provided unit '%s'.", tab_unit, cm_unit)
+            log.debug(f"Attempting auto-convert of table unit {tab_unit} to "
+                      f"user-provided unit {cm_unit}.")
 
             if not isinstance(cm_unit, u.Unit):
                 cm_unit = u.Unit(cm_unit)
-            if cm_unit.physical_type in ('length', 'frequency'):
+            cm_type =  str(cm_unit.physical_type)
+            if 'length in cm_type' or 'frequency' in cm_type or 'energy' in cm_type:
```

Comment:
```suggestion
            if 'length' in cm_type or 'frequency' in cm_type or 'energy' in cm_type:
```
Looks like one of the quotes is misplaced here.

Reply:
Thanks, yes; I should stay in Emacs+Flycheck for this stuff.